heartbeat/CTDB.in

```diff
@@ -589,17 +580,20 @@ ctdb_start() {
 	local log_option
 	# --logging supported from v4.3.0 and --logfile / --syslog support 
 	# has been removed from newer versions
-	version=$(ctdb version | awk '{print $NF}')
-	ocf_version_cmp "$version" "4.2.14"
-	if [ "$?" -eq "2" ]; then
-		log_option="--logging=file:$OCF_RESKEY_ctdb_logfile"
+	version=$(ctdb version | awk '{print $NF}' \
+					| awk -F "." '{print $1 "." $2 }')
```

Comment:
You should use 
sed "s/[[:alpha:]]\+.*//"
instead.

That way you can compare to the full version still.

Reply:
That'd produce suboptimal output for version strings which lack a revision suffix but include a vendor suffix, e.g.
 <pre>
# echo 4.10.0.GIT.2d71ec09c32 | sed "s/[[:alpha:]]\+.*//"
4.10.0.
</pre>

I'd like to avoid yet another change + test + submit cycle here if possible. Besides, there's no reason to consider anything aside from the major and minor version numbers - 4.3.0 is the cutoff.

Reply:
This solves that:
sed "s/\.*[[:alpha:]]\+.*//"

Reply:
Okay, given that you seem to be insisting that I use sed here, I'll need to drop this for now until I have time to retest.

Reply:
Ah. That looks good to me.

With the awk-variant you cant check the full version-string, so you have to change the version number you're checking against and rc from the comparison, which I'd prefer not to do in case there's a later sub-version where it works the new way.

Reply:
You can remove the "\+", as .* matches that too.

Reply:
I've dropped the entire change for now. I'll resubmit the https://github.com/ClusterLabs/resource-agents/issues/1185 fix later, when I have time to retest.
Please pull v4.